### PR TITLE
[VL] Add RowToColumnarExec to FakeRowAdaptor when use velox native parquet write

### PR DESCRIPTION
**Pull request by WolverineJiang**
## What changes were proposed in this pull request?

Some problems bring in by patch #1925 . This patch will resolve it.

## How was this patch tested?

Run UT by set spark.gluten.sql.native.writer.enabled=true

Run TPCDS by set spark.gluten.sql.native.parquet.writer.enabled=true && spark.sql.sources.useV1SourceList=parquet

### CODE REVIEW

File: gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala [hunk 1]
```diff
@@ -194,7 +194,15 @@ object GlutenWriterColumnarRules {
...
                     ))))
-            case other => rc.withNewChildren(Array(FakeRowAdaptor(other)))
+            case other =>
+              if (other.supportsColumnar) {
+                rc.withNewChildren(Array(FakeRowAdaptor(other)))
+              } else {
+                rc.withNewChildren(
+                  Array(
+                    FakeRowAdaptor(
+                      BackendsApiManager.getSparkPlanExecApiInstance.genRowToColumnarExec(other))))
```
Comment (PHILO-HE) [inline gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala:204]: Thanks for your PR!
I note `doExecuteColumnar` at `FakeRowAdaptor` has a similar handling for row-based child operator. Doesn't that piece of code work?
If there is any issue found in your test, could you please add some UTs? Thanks!

File: gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala [hunk 2]
```diff
@@ -194,7 +194,15 @@ object GlutenWriterColumnarRules {
                       aqe.isSubquery,
                       supportsColumnar = true
                     ))))
-            case other => rc.withNewChildren(Array(FakeRowAdaptor(other)))
+            case other =>
+              if (other.supportsColumnar) {
+                rc.withNewChildren(Array(FakeRowAdaptor(other)))
```
Comment (jackylee-ch) [inline gluten-core/src/main/scala/org/apache/spark/sql/execution/datasources/GlutenWriterColumnarRules.scala:199]: Please add  some UTs to show why we should do this and confirm problem fixed.

**Comment by jackylee-ch** [comment]
@PHILO-HE I just checked the problem @WolverineJiang  met. it has nothing to do with the current pr, he did not override the `ParquetFileFormat` through `extraClassPath`. Thus I will close this pr now.

**Comment by PHILO-HE** [comment]
> @PHILO-HE I just checked the problem @WolverineJiang met. it has nothing to do with the current pr, he did not override the `ParquetFileFormat` through `extraClassPath`. Thus I will close this pr now.

No problem. Thanks for your clarification!

**Pull request closed without merging**